### PR TITLE
batch send tx

### PR DESCRIPTION
**Pull request by juniuszhou**
The pr is to send the transaction to sifnode with limited events. From the load testing, we find out the error about message too large.

### CODE REVIEW

**Comment by jzvikart** [comment]
Interesting... I suddenly remembered that I was seeing similar problems when trying to do transfers of multiple denoms between sif accounts in a single transaction. Transactions with just a few denoms would work and more than just a few would not. Funny thing is that I did **exactly the same thing** in siftool - chopped up transfers into batches of 5. This makes me wonder if we're not seeing the same issue here.

**Comment by juniuszhou** [comment]
> Interesting... I suddenly remembered that I was seeing similar problems when trying to do transfers of multiple denoms between sif accounts in a single transaction. Transactions with just a few denoms would work and more than just a few would not. Funny thing is that I did **exactly the same thing** in siftool - chopped up transfers into batches of 5. This makes me wonder if we're not seeing the same issue here.

Actually, I didn't reproduce the same load testing. From the source code, I can see that currently, relayer put all events in a Ethereum block into single message, then sent to Sifnode. If too many events, then the message could be very large. Even without the error found in the load testing, I think we still can do this code change to optimize the message sending in relayer.

File: cmd/ebrelayer/txs/relayToCosmos.go [hunk 1]
```diff
@@ -41,18 +41,41 @@ func RelayToCosmos(factory tx.Factory, claims []*ethbridge.EthBridgeClaim, cliCt
 			continue
 		} else {
 			messages = append(messages, &msg)
+			// to avoid too many data in single transaction, send out by batch
+			if len(messages) == 5 {
```
[the remark this answers is not in the document]
Comment (smartyalgo) [reply cmd/ebrelayer/txs/relayToCosmos.go:45]: Can you add unit test for when claims size is 
1. 0
2. < 5
3. == 5
4.  \>  5

**Review by smartyalgo — COMMENTED** [review]
This looks good to me, minor comments

File: cmd/ebrelayer/txs/relayToCosmos.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToCosmos.go:45]: will define a variable.

File: cmd/ebrelayer/txs/relayToCosmos.go [hunk 2]
```diff
@@ -41,18 +41,41 @@ func RelayToCosmos(factory tx.Factory, claims []*ethbridge.EthBridgeClaim, cliCt
...
+		sugaredLogger.Infow("RelayToCosmos building, signing, and broadcasting", "messages", messages)
+		instrumentation.PeggyCheckpointZap(sugaredLogger, "BroadcastTx", zap.Reflect("messages", messages))
+		return err
+	}
 
+	return nil
+}
+
+// Send the messages to sifnode via broadcast transaction
+func sendMessagesToCosmos(factory tx.Factory, cliCtx client.Context, messages *[]sdk.Msg, sugaredLogger *zap.SugaredLogger) error {
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply cmd/ebrelayer/txs/relayToCosmos.go:69]: not necessary for slice in golang, will update it.

[the remark this answers is not in the document]
Comment (smartyalgo) [reply cmd/ebrelayer/txs/relayToCosmos.go]: In our case it's not necessary, but heres a good read too https://medium.com/swlh/golang-tips-why-pointers-to-slices-are-useful-and-how-ignoring-them-can-lead-to-tricky-bugs-cac90f72e77b

**Comment by pandaring2you** [comment]
jure already merged these changes with a different PR

**Comment by juniuszhou** [comment]
close it since the code change already merged in future/peggy2

**Pull request closed without merging**